src/Uno.UWP/Networking/Connectivity/ConnectionProfile.Android.cs

```diff
@@ -0,0 +1,78 @@
+﻿#if __ANDROID__
+using System;
+using System.Linq;
+using AndroidConnectivityManager = Android.Net.ConnectivityManager;
+
+namespace Windows.Networking.Connectivity
+{
+
+	public partial class ConnectionProfile
+	{
+		private AndroidConnectivityManager _connectivityManager;
+
+		internal ConnectionProfile(bool isInternet)
+		{
+			// using such constructor, we can fill data only when getting object from Windows.Networking.Connectivity.NetworkInformation.GetInternetConnectionProfile()
+
+			// test Manifest (Access_Network_State)
+			var context = Android.App.Application.Context;
+			var packageInfo = context.PackageManager.GetPackageInfo(
+									context.PackageName, Android.Content.PM.PackageInfoFlags.Permissions);
+			var requestedPermissions = packageInfo?.RequestedPermissions;
+			if (requestedPermissions is null)
+			{
+				throw new UnauthorizedAccessException("no ACCESS_NETWORK_STATE permission defined in Manifest");
+			}
+
+			bool bInManifest = requestedPermissions.Any(p => p.Equals(Android.Manifest.Permission.AccessNetworkState, StringComparison.OrdinalIgnoreCase));
+			if (!bInManifest)
+			{
+				throw new UnauthorizedAccessException("no ACCESS_NETWORK_STATE permission defined in Manifest");
+			}

```

Comment:
Use the method that was added in `PermissionHelper`

Reply:
I actually didn't do refactoring of the code before the review, now it is changed.

Reply:
Speaking of permissionHelper... If you could help me with it :)
I created methods with string[] on input (#3025), without need to switch between Uno.UI and Uno.UWP. But it was rejected; all code have to be moved into Uno.UI... But I don't know how to add another delegate methods. In normal C, I would use pointer to method :)
I need two methods inside Uno.UI PermissionHelper (e.g.  `public static string[] MissingPermissions(string[] requiredPermissions, string[] optionalPermissions)` and `public static Task<bool> TryGetPermissionAsync(string[] requiredPermissions, string[] optionalPermissions, bool ignoreErrors = false)` to be callable from Uno.UWP.
Probably there should two more parameters here:
https://github.com/unoplatform/uno/blob/6e8a9273acbd6fcab385fc36a3d29cb48295f824/src/Uno.UI/Extensions/PermissionsHelper.Android.cs#L22, but simply I don't know how to fit what I want into C# syntax :)

Reply:
Or maybe I already know it :)

Reply:
Sorry, missed this one, have you been able to resolve this?

Reply:
See #3025 .
You can use MissingPermission(your_permission) to test if permission is granted, and TryAskPermissionAsync(your_permission) if you want to ask user for permission (system dialog). These methods can accept also string[] (if you want to test for many permissions at once), and some more functionality :)
It was tested in previous versions, when whole code resides in Uno.UWP (and is used in some of my apps), now code is moved to Uno.UI with call from Uno.UWP (as requested by Jerome) using complicated delegates; not tested in app.